build-tools/packages/version-tools/src/test/versions.test.ts

```diff
@@ -159,3 +159,145 @@ describe("getIsLatest", () => {
 		assert.isFalse(getIsLatest("client", "1.2.3", post1_tags, true));
 	});
 });
+
+describe("legacy compat ranges", () => {
+	it("legacy compat: 2.0.9 and compat version interval 10", () => {
+		const input = `2.0.9`;
+		const expected = `>=2.0.9 <2.10.0`;
+		const range = generateLegacyCompatRange(input, 10);
+		assert.strictEqual(range, expected);
+	});
+
+	it("legacy compat: 2.8.10 and compat version interval 10", () => {
+		const input = `2.8.10`;
+		const expected = `>=2.8.10 <2.10.0`;
+		const range = generateLegacyCompatRange(input, 10);
+		assert.strictEqual(range, expected);
+	});
+
+	it("legacy compat: 2.0.9 and compat version interval 20", () => {
+		const input = `2.0.9`;
+		const expected = `>=2.0.9 <2.20.0`;
+		const range = generateLegacyCompatRange(input, 20);
+		assert.strictEqual(range, expected);
+	});
+
+	it("legacy compat: 2.8.10 and compat version interval 20", () => {
+		const input = `2.8.10`;
+		const expected = `>=2.8.10 <2.20.0`;
+		const range = generateLegacyCompatRange(input, 20);
+		assert.strictEqual(range, expected);
+	});
+
+	it("legacy compat: 2.18.10 and compat version interval 20", () => {
+		const input = `2.18.10`;
+		const expected = `>=2.18.10 <2.20.0`;
+		const range = generateLegacyCompatRange(input, 20);
+		assert.strictEqual(range, expected);
+	});
```

Comment:
Don't know that you need all of those compat range cases.
But it is good to have more of the same input versions with the various intervals. 2.18.10 would be good to test with interval 10 as well.
When there is a combo of things it is cool if there is a test matrix - preferably easy to read.
I think this is such a thing:
```typescript
	const testMatrix = [
		["2.0.9", {10: "2.10.0", 20: "2.20.0", 25: "2.25.0", 30: "2.30.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
		["2.8.10", {10: "2.10.0", 20: "2.20.0", 25: "2.25.0", 30: "2.30.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
		["2.18.10", {10: "2.20.0", 20: "2.20.0", 25: "2.25.0", 30: "2.30.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
		["2.0.10", {10: "2.10.0", 20: "2.20.0", 25: "2.25.0", 30: "2.30.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
		["2.10.0", {10: "2.10.0", 20: "2.20.0", 25: "2.25.0", 30: "2.30.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
		["2.25.0", {10: "2.30.0", 20: "2.30.0", 25: "2.25.0", 30: "2.30.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
		["2.39.10", {10: "2.40.0", 20: "2.40.0", 25: "2.50.0", 30: "2.60.0", 40: "2.40.0", 50: "2.50.0", 200: "2.200.0"}],
	];
	for (const [inputVersion, subCases] of testMatrix) {
		for (const [interval, upperBound] of Object.entries(subCases)) {
			it(`legacy compat: ${inputVersion} and compat version interval ${interval} yields ">=${inputVersion} <${upperBound}"`, () => {
				const range = generateLegacyCompatRange(inputVersion, parseInt(interval, 10));
				assert.strictEqual(range, `>=${inputVersion} <${upperBound}`);
			});
		}
	}
```
It covers more cases. Skips just a few.
The test names are also very clear. From the run you know exactly what is tested and expectations. Like a spec.
I don't think Biome likes the formatting like that - maybe it's preference is also easy to read.